### PR TITLE
Fix commenting

### PR DESCRIPTION
**Pull request by ryzokuken**
Make sure these boxes are checked before your pull request is ready to be reviewed and merged. Thanks!

* [ ] all tests pass -- `rake test:all`
* [ ] code is in uniquely-named feature branch, and has been rebased on top of latest master (especially if you've been asked to make additional changes)
* [ ] pull request is descriptively named with #number reference back to original issue

Please be sure you've reviewed our contribution guidelines at https://publiclab.org/wiki/contributing-to-public-lab-software

We have a loose schedule of reviewing and pulling in changes every Tuesday and Friday, and publishing changes on Fridays. Please alert developers on plots-dev@googlegroups.com when your request is ready or if you need assistance.

Thanks!

### CODE REVIEW

**Comment by ryzokuken** [comment]
@jywarren @ebarry this is how commenting looked locally 🎉 

screenshot from 2017-08-03 01-21-53

**Comment by ryzokuken** [comment]
@jywarren This will most likely fail, as the tests were written keeping the faulty code in mind. I will fix those tomorrow if that's not a problem, Thanks.

**Comment by ananyo2012** [comment]
@ryzokuken This looks great! Did you use curl for making the post request?

**Comment by ryzokuken** [comment]
No, I was using Firefox's http requester add-on. You could also use Chrome
if you're using Chrome.

**Comment by ananyo2012** [comment]
Well I mostly Use Postman plugin for making post requests.

**Comment by ryzokuken** [comment]
@jywarren @ananyo2012 There's an issue. Until now, I had been adding stuff to the session in the request, and had been reading from the session itself. In order to fix this, we could make an integration test instead, as I had been confused between http://api.rubyonrails.org/v3.2/classes/ActionController/TestCase/Behavior.html#method-i-post and http://api.rubyonrails.org/v3.2/classes/ActionDispatch/Integration/RequestHelpers.html#method-i-post. What do you guys say?

**Comment by jywarren** [comment]
Oh huh... just curious, why do you use the session?

**Comment by ryzokuken** [comment]
I was confused between the two, and set the session instead of the header.

**Comment by ananyo2012** [comment]
Well we can simply use `params[:token]` and send a simple key-value pair for the token via a post request. It doesn't necessarily need to be in the header. Though the convention for token based authentication is sending the token in an `Authorization` header.

File: app/controllers/comment_controller.rb [hunk 1]
```diff
@@ -44,7 +44,7 @@ def create_by_token
     @node = Node.find params[:id]
     @user = User.find_by_username params[:username]
     @body = params[:body]
-    @token = request.env['rack.session']['token']
+    @token = request.headers["HTTP_TOKEN"]
```
Comment (ananyo2012) [inline app/controllers/comment_controller.rb]: @ryzokuken Can you check this with `@request.headers["HTTP_TOKEN"]` and check if the tests pass.

**Comment by ryzokuken** [comment]
Exactly. I don't think sending the token as a parameter is safe at all.

**Comment by ryzokuken** [comment]
It's being read perfectly, as comments work. The tests are configured
improperly. There's no way to set headers inside controller tests, you can
only set headers in request inside integration tests.

**Comment by ananyo2012** [comment]
So what is assigned to the `@token` during tests ? Is it null ?

**Comment by ryzokuken** [comment]
@ananyo2012 yes.

**Comment by jywarren** [comment]
Ah, i see. Well, yes, if you are confident that the header is the appropriate place (and worth the extra complexity for users of the API; consider how difficult it is to set a header from any particular scripting tool), then go ahead and move this to be an integration test. Thanks!

**Comment by ryzokuken** [comment]
@jywarren sure, thanks! I will submit a PR that moves the tests to an integration test instead today.

**Comment by ryzokuken** [comment]
@jywarren @ananyo2012 yay! This looks ready from my side.

**Comment by ryzokuken** [comment]
@jywarren @icarito Let me know when these changed are pushed to staging and stable, then I could proceed with using these.

**Comment by jywarren** [comment]
It is live now on PublicLab.org!

**Comment by ryzokuken** [comment]
@jywarren @ebarry @ananyo2012 demonstrated this on the live site, it works!

**Comment by jywarren** [comment]
This is so super!

**Pull request merged**